geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DeployWithGroupsDUnitTest.java

```diff
@@ -200,6 +199,67 @@ public void deployJarToAllServers() throws Exception {
     server2.invoke(() -> assertThatCanLoad(jarName1, class1));
   }
 
+  @Test
+  public void deployJarToAllServersWithRestart() throws Exception {
+    // Deploy a jar to all servers
+    gfshConnector.executeAndAssertThat("deploy --jar=" + jar1).statusIsSuccess();
+    String resultString = gfshConnector.getGfshOutput();
+
+    assertThat(resultString).contains(server1.getName());
+    assertThat(resultString).contains(server2.getName());
+    assertThat(resultString).contains(jarName1);
+
+    server1.invoke(() -> assertThatCanLoad(jarName1, class1));
+    server2.invoke(() -> assertThatCanLoad(jarName1, class1));
+
+    server1.getVM().bounce();
```

Comment:
does bounce cleans the filesytem as well? can we use server1.stopVM(false)?

Reply:
`bounce()` does not clean the filesystem AFAICT.